lib/clients.ts

```diff
@@ -0,0 +1,113 @@
+import { Consumer, Producer, Kafka, EachMessagePayload, logLevel } from 'kafkajs';
+
+class KafkaClient {
+  bootstrapServer: string;
+  username: string;
+  password: string;
+  kafka: Kafka;
+
+  constructor(bootstrapServer: string, username: string, password: string) {
+    this.bootstrapServer = bootstrapServer;
+    this.username = username;
+    this.password = password;
+    this.kafka = new Kafka({
+      brokers: [bootstrapServer],
+      ssl: true,
+      sasl: {
+        // TODO add oauth
+        mechanism: 'plain',
```

Comment:
Both configurations are available to the users so I think we should cover both.

I think it depends if we want to I am to merge this PR before Christmas or not. I will be off from next Tuesday. However, I don't think the change will be too hard so I will check it today, and maybe we can do a final decision later today or tomorrow.?

Reply:
I have been a little bit busier than I expected. I played a little bit with it and decided to implement it in the next PR to avoid additional conflicts and delays with this. Opened https://issues.redhat.com/browse/MGDSTRM-10308 .

Reply:
ok 👍